src/components/ProductCard/components/CartButton.jsx

```diff
@@ -0,0 +1,42 @@
+import { useLocation, useNavigate } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { useAuth, useCart } from "../../../context";
+
+export const CartButton = ({ product }) => {
+  const { authData } = useAuth();
+  const { cartList, handleCartAddition } = useCart();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [buttonText, setButtonText] = useState("Add to cart");
+
+  useEffect(() => {
+    if (cartList.find((item) => item._id === product._id)) {
+      location.pathname === "/wishlist"
+        ? setButtonText("Already in cart")
+        : setButtonText("Go to Cart");
+    } else {
+      setButtonText("Add to cart");
+    }
+  }, [cartList]);
+
+  return (
+    <button
+      className={`btn btn--primary w-100 ${
+        buttonText === "Adding" || buttonText === "Already in cart"
```

Comment:
Everything looks good but here I guess You are not allowing the user to add more than one items to cart from the productListing Page. If that is the case it might be a bad UX .

Reply:
Thanks, @aniketxparihar  for the review. No, Actually you can add only one quantity for each item in the product page.